crates/tasks_ui/src/lib.rs

```diff
@@ -31,8 +31,11 @@ pub fn init(cx: &mut AppContext) {
                             project.task_inventory().read(cx).last_scheduled_task()
                         })
                     {
-                        if action.reevaluate_context {
-                            let mut original_task = last_scheduled_task.original_task().clone();
+                        let mut original_task = last_scheduled_task.original_task().clone();
+                        if let Some(reevaluate_context) = action.reevaluate_context {
+                            original_task.reevaluate_context = reevaluate_context;
```

Comment:
This seems dangerous and most probably will break: `original_task` will be resolved and stored into history, now with `reevaluate_context = true` which might not be the default case in the original task template.

This is important, as here it's the `action` that forces the task to reevaluate its context, not the task per se, and future rerun tasks that do not force the context reevaluation, should pick up the old context from the resolved task.